### DISCUSSION TITLE
Cache updates with nested queries

### DISCUSSION THREAD

**Discussion by smontlouis** [urql-graphql/urql#2451] [Q&A] [answered]
I'm a little confused about how to make cache updates work in this case.
I have the following :

```gql
favorites(filters: { user_id: { id: { eq: $userId } } }) {
    books (filters: { some_filters }) {

    }
    comics (filters: { some_filters }) {
    }
}
```

Everytime I update favorite books, I want to update comic books:

```js
// exchanges -> cacheExchange -> updates
updateUserFavorite: (result, args, cache, info) => {
    cache
        .inspectFields('Query')
        .filter((field) => field.fieldName === 'favorites')
        .forEach((field) => {
            cache.updateQuery(
                {
                query: GET_FAVORITES,
                variables: {...},
                },
                data => { // do some stuff with the data }
                )
            )
}
```

Here `field` only have `filters.user_id.id.eq` as arguments, I'm not sure where to find book filters and comic filters in order to update the cache.

**Comment by JoviDeCroock** [comment, marked as answer]
This question is a bit broad, what are you trying to do i.e. if you just want to update the books that are favorited i.e. append them to the list I would doubt it would be easy to do for all filters as you would then have to mirror the logic from your back-end in terms of filtering which isn't really feasible imho. If your only filter for favorites is the user_id that could be in scope if you pass it into the mutation by means of `info.variableValues` or you can search for it by going through all the available book0filters you have. A lot of this depends however on the arguments that you have available to you.

In cases where updating a list is hard I often just say use `invalidate`

**Reply by smontlouis** [reply]
You made a valid point. I believe this is not the best architecture choice regarding the database.
Even when we think about returning the updated object, we need to filter again.

Instead of this, we will go with
`favorite_books`
`favorite_comics`

Much less headache.


Thank you !
